compiler/rustc_lint/src/context.rs

```diff
@@ -404,6 +404,7 @@ impl LintStore {
             // FIXME: rustc and rustdoc are considered tools for lints, but not for attributes.
             if tool_name != sym::rustc
                 && tool_name != sym::rustdoc
+                && tool_name != sym::diagnostic
```

Comment:
Why is this required?
We don't have `diagnostic::foo` lints yet, only attributes.

Reply:
I've just blindly added this kind of check everywhere were existing checks for the other 2 built-in tool modules existed. If that is not required here I'm happy to remove it.